name: Test
on: [push, pull_request]

jobs:
  build:
    runs-on: ubuntu-latest
    services:
      postgres:
        image: postgres:12
        env:
          POSTGRES_USER: test
          POSTGRES_PASSWORD: test
          POSTGRES_DB: 360givingdatastore
        options: >-
          --health-cmd pg_isready
          --health-interval 10s
          --health-timeout 5s
          --health-retries 5
        ports:
          - 5432:5432/tcp
    steps:
    - uses: actions/checkout@v2
    - name: Setup python
      uses: actions/setup-python@v2
      with:
        python-version: 3.8
        architecture: x64
    - uses: actions/cache@v1
      with:
        path: ~/.cache/pip
        key: ${{ runner.os }}-pip-${{ hashFiles('**/requirements.txt') }}-${{ hashFiles('**/requirements_dev.txt') }}

    # For generating random data
    - run: sudo apt-get install pwgen

    - run: pip install -r requirements.txt

    # Check all apps load correctly w/o dev requirements
    - run: ./datastore/manage.py --help
    - run: ./datastore/manage.py check

    - run: pip install -r requirements_dev.txt
    # FIXME manual install datagetter requirements
    - run: pip install -r src/datagetter/requirements.txt

    - run: ./datastore/manage.py makemigrations --dry-run --check
    - run: ./datastore/manage.py migrate
    - run: coverage run --source='./datastore' ./datastore/manage.py test -v 3 tests
      env:
        DATA_RUN_PID_FILE: datarunpidfile.pid
        DJANGO_SETTINGS_MODULE: settings.settings_test
    - name: Verify it compiles and is installed
      run: datagetter.py --help

    - env:
        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
      run: coveralls --service=github
